tracer/src/Datadog.Trace.SourceGenerators/Helpers/EquatableArray.cs

```diff
@@ -20,7 +20,7 @@ namespace Datadog.Trace.SourceGenerators.Helpers;
     /// <summary>
     /// The underlying <typeparamref name="T"/> array.
     /// </summary>
-    private readonly T[]? _array;
+    internal readonly T[]? _array;
```

Comment:
We have `AsArray()` for accessing this already 😉 
```suggestion
    private readonly T[]? _array;
```

Reply:
Fixed in dbc916e706f23dd123a861af457b68f4c5ab5c5e